src/loader/scene.rs

```diff
@@ -0,0 +1,217 @@
+    })?;
+
+    let mut mat_handles = Vec::with_capacity(materials.len());
+    for (mat_idx, mat) in materials.into_iter().enumerate() {
+        let mut material = StandardMaterial {
+            base_color_texture: load_mat_texture(
+                &mat.diffuse_texture,
+                load_context,
+                supported_compressed_formats,
+            )
+            .await?,
+            normal_map_texture: load_mat_texture(
+                &mat.normal_texture,
+                load_context,
+                supported_compressed_formats,
+            )
+            .await?,
+            ..Default::default()
+        };
+        if let Some(color) = mat.diffuse {
+            material.base_color = Color::rgb(color[0], color[1], color[2]);
+        }
+        mat_handles.push(
+            load_context.set_labeled_asset(&material_label(mat_idx), LoadedAsset::new(material)),
+        );
+    }
+
+    let mut world = World::default();
+    let world_id = world.spawn(SpatialBundle::INHERITED_IDENTITY).id();
+    for (model_idx, model) in models.into_iter().enumerate() {
+        let vertex_position: Vec<[f32; 3]> = model
+            .mesh
+            .positions
+            .chunks_exact(3)
+            .map(|v| [v[0], v[1], v[2]])
+            .collect();
+        let vertex_normal: Vec<[f32; 3]> = model
+            .mesh
+            .normals
+            .chunks_exact(3)
+            .map(|n| [n[0], n[1], n[2]])
+            .collect();
+        let vertex_texture: Vec<[f32; 2]> = model
+            .mesh
+            .texcoords
+            .chunks_exact(2)
+            .map(|t| [t[0], 1.0 - t[1]])
+            .collect();
+
+        let mut mesh = Mesh::new(PrimitiveTopology::TriangleList);
+        mesh.set_indices(Some(Indices::U32(model.mesh.indices)));
+
+        mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, vertex_position);
+        if !vertex_texture.is_empty() {
+            mesh.insert_attribute(Mesh::ATTRIBUTE_UV_0, vertex_texture);
+        }
+
+        if !vertex_normal.is_empty() {
+            mesh.insert_attribute(Mesh::ATTRIBUTE_NORMAL, vertex_normal);
+        } else {
+            mesh.duplicate_vertices();
+            mesh.compute_flat_normals();
+        }
+
+        let mesh_handle =
+            load_context.set_labeled_asset(&mesh_label(model_idx), LoadedAsset::new(mesh));
+
+        // Now assign the material
+        let pbr_id = if let Some(mat_id) = model.mesh.material_id {
+            world
+                .spawn(PbrBundle {
+                    mesh: mesh_handle,
+                    material: mat_handles[mat_id].clone(),
+                    ..Default::default()
+                })
+                .id()
+        } else {
+            world
+                .spawn(PbrBundle {
+                    mesh: mesh_handle,
+                    ..Default::default()
+                })
+                .id()
+        };
+        world.entity_mut(world_id).push_children(&[pbr_id]);
```

Comment:
Do we need this parent entity? To my understanding the `SceneBundle` used to spawn this "world" will handle creating a parent entity.

Reply:
Good call, to be honest I did it out of habit for how visibility and hierarchy works in normal applications. I removed all the manual hierarchy updates in https://github.com/AmionSky/bevy_obj/pull/12/commits/155dc8098972ec3045445282a20dbc233c946280